src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -150,6 +150,7 @@ private static IArgument DeriveArgument(int parameterIndex, int argumentIndex, I
                 (argument) =>
                 {
                     string name = !argumentNames.IsDefaultOrEmpty ? argumentNames[argumentIndex] : null;
+                    Symbols.ParameterSymbol parameter = parameterIndex < parameters.Length ? parameters[parameterIndex] : null;
```

Comment:
Should we bail out at the start of the method and return `boundArguments[argumentIndex]` if `HasErrors` is true? I presume we cannot guarantee a correct argument to parameter mapping in presence of errors.